runtime/compiler/runtime/JITClientSession.cpp

```diff
@@ -666,14 +667,16 @@ void ClientSessionData::purgeCache(std::vector<ClassUnloadedData> *unloadedClass
    }
 
 void
-ClientSessionData::readAcquireClassUnloadRWMutex()
+ClientSessionData::readAcquireClassUnloadRWMutex(TR::CompilationInfoPerThreadBase *compInfoPT)
    {
+   static_cast<TR::CompilationInfoPerThreadRemote *>(compInfoPT)->incrementClassUnloadReadMutexDepth();
```

Comment:
I would prefer to increment the depth after acquiring the mutex. Otherwise, when the thread is forced to wait because the mutex is held for writing, there will be a rather large window of time when the counter is misleading. This is probably only relevant for debugging though if a thread increments and decrements its own counter.
We should probably have some comment saying that for these routines, `compInfoPT` received as parameter must be associated with the current executing thread.